lib/src/multires_stochastic_texture_synthesis.rs

```diff
@@ -533,10 +533,16 @@ impl Generator {
         }
 
         let mut rng = Pcg32::seed_from_u64(m_seed);
-
         //random candidates
         for _ in 0..m_rand {
-            let rand_map = (rng.gen_range(0, example_maps.len())) as u32;
+            let mut rand_map = (rng.gen_range(0, example_maps.len())) as u32;
```

Comment:
The problem was if we chose a map which was ignored, we'd keep looping forever here https://github.com/EmbarkStudios/texture-synthesis/blob/4b1ec51ddce162ffc51b8b22e5f2427317dd3121/lib/src/multires_stochastic_texture_synthesis.rs#L549 because no part of the image would be valid

Reply:
Also I'm wondering if it may be better to change this whole method to sample from a list of valid locations across images rather than the current draw / check loops. I think in the case of large images with tiny valid sample masks the current method could take a while to find a valid pixel. But I'm not sure how common that type of case is and this is the smallest code change I could make.

Reply:
Yes, I think filtering out the ignore images once at the beginning of the resolve loop makes the most sense.

Reply:
👍